app/scripts/metamask-controller.js

```diff
@@ -4831,6 +4832,31 @@ export default class MetamaskController extends EventEmitter {
     }
   }
 
+  /**
+   * Used to create a CAIP stream for connecting to an untrusted context.
+   *
+   * @param options - Options bag.
+   * @param {ReadableStream} options.connectionStream - The Duplex stream to connect to.
+   * @param {MessageSender | SnapSender} options.sender - The sender of the messages on this stream.
+   * @param {string} [options.subjectType] - The type of the sender, i.e. subject.
+   */
+
+  setupUntrustedCommunicationCaip({ connectionStream, sender, subjectType }) {
+    let _subjectType;
```

Comment:
Nit: I know a leading underscore is commonly used like this to avoid shadowing, but it's a bit confusing because of the other convention we have where a leading underscore indicates an unused parameter. It sorta defeats the purpose of avoiding shadowing in general (that rule is meant to prevent confusing two variables, but here we might still mistake them for each other because of how similar the names are)

Perhaps we can alias the parameter instead? That would allow us to give it a more meaningful name to ensure it isn't used anywhere instead of `_subjectType`. e.g. we could call it `inputSubjectType`.

Reply:
good point. updated usages here https://github.com/MetaMask/metamask-extension/pull/25075/commits/0c3f000f2150e6fe870c91df20166b3eda9ad436